### ISSUE TITLE
Update to v1.2.0

### ISSUE THREAD

**Issue by agardnerIT** [afzal442/keptn-pg-killerway#74]
Update killercoda demo to use Keptn v1.1.0

**Comment by Amishakumari544** [comment]
I am interested to work on this @agardnerIT how can I start working on this?

**Comment by rakshitgondwal** [comment]
I would like to work on this too! @agardnerIT

**Comment by agardnerIT** [comment]
Excellent!

1. Fork this repo
2. Sign in to killercoda.com with GitHub and set up the sync for your forked repo [as per their instructions](https://killercoda.com/creators)
3. You will get a duplicate killercoda scenario under your account eg. `https://killercoda.com/YOU`

Due to the webhook that you need to set up in step 2 (follow the instructions), Killercoda will automatically sync whenever you change your fork.

Update the scenario to use Keptn v1.1.0 rather than v1.0.0 and when you've tested it, raise a PR back here = all done.

I'll let you co-ordinate between yourselves on how / when best to work :)

Any problems, let me know!

**Comment by agardnerIT** [comment]
Start with something small like changing a word in one of the markdown files. Killercoda should see the changes on main and "auto-refresh" the tutorial. You'll see the spinner on Killercoda.com and if that works, you know the mechanics are working correctly.

Then, it should just be a case of changing this line: https://github.com/afzal442/keptn-pg-killerway/blob/877034701854b453e9a126155419e8fe7b2a14ae/keptn-end-to-end-delivery/intro_foreground.sh#L8 so it matches the version number of the latest release (eg. `1.1.0`)

![image](https://user-images.githubusercontent.com/26523841/217682298-7e8ebe21-4556-4faf-a220-78a88aed9e4e.png)

Hint: You might want to increment the `DEBUG_VERSION` just so when you refresh the Killercoda site you know it actually has been updated - that's all I use that field for.

Run through the tutorial and validate it works then raise the PR :)

**Comment by rakshitgondwal** [comment]
@agardnerIT Everything is working perfectly untill the step 3.
I am facing issues in step 4, i.e the automated go/non-go decision step using prometheus.
I am getting this `502 bad gateway-nginx` error. Can you help me out with this.

Here are some screenshots of the error I faced- 
![Screenshot from 2023-02-09 18-41-24](https://user-images.githubusercontent.com/98955085/217825359-c1b75143-7030-48fe-8ed6-093d07c91ff0.png)
![Screenshot from 2023-02-09 18-52-04](https://user-images.githubusercontent.com/98955085/217825314-c6ced1da-e508-4bac-b5a1-95bd44571cee.png)

**Comment by rakshitgondwal** [comment]
Also, I didn't get you with the `DEBUG_VERSION` thing. Why do we need to increment it?
I did increment it to 2 and it is working perfectly fine but I didn't understood the use of it.

**Comment by agardnerIT** [comment]
What is your killercoda.com account so I can run through the tutorial. Also where is the repo that you're using for Killercoda?

`DEBUG_VERSION` really has only one use. On killercoda.com when you access a tutorial, it's hard to tell whether it has been updated or not. I use DEBUG_VERSION just to make it easier for me to see that the update has "taken effect" and killercoda is actually service my latest version. That's all. It has no real use. It's just for us as developers.

**Comment by agardnerIT** [comment]
[1.2.0 has been released](https://github.com/keptn/keptn/releases). Let's use this opportunity to move straight to that version?

If you agree, I'll change the title of this issue.

**Comment by rakshitgondwal** [comment]
> What is your killercoda.com account so I can run through the tutorial. Also where is the repo that you're using for Killercoda?
> 
> `DEBUG_VERSION` really has only one use. On killercoda.com when you access a tutorial, it's hard to tell whether it has been updated or not. I use DEBUG_VERSION just to make it easier for me to see that the update has "taken effect" and killercoda is actually service my latest version. That's all. It has no real use. It's just for us as developers.

[Killercoda account](https://killercoda.com/rakshit)
[Killercoda repo](https://github.com/rakshitgondwal/keptn-pg-killerway). I am using dev1 branch for the syncing.

**Comment by rakshitgondwal** [comment]
> [1.2.0 has been released](https://github.com/keptn/keptn/releases). Let's use this opportunity to move straight to that version?
> 
> If you agree, I'll change the title of this issue.

Wow amazing!! Sure you can change the title of the issue. I'll also try using the tutorial with `1.2.0` version now.

**Comment by rakshitgondwal** [comment]
I was trying to follow the tutorial with version `1.2.0` but I am getting an error in the installation stap itself @agardnerIT 
![Screenshot from 2023-02-10 21-01-33](https://user-images.githubusercontent.com/98955085/218136533-77910759-efad-4286-8e28-d62e1dabc0dc.png)

**Comment by agardnerIT** [comment]
I think something went wrong with the v1.2.0 release. It only includes 2 files rather than the 26 for all other releases - so the Killercoda install fails. I don't think it is anything you've done.

Either they've changed the way they release or it is actually broken. I've posted in Keptn slack. Let's see what come back. We're blocked until we get an answer.

The error on this line of Killercoda gave me the hint:

```
Error: INSTALLATION FAILED: failed to fetch https://github.com/keptn/keptn/releases/download/1.2.0/keptn-1.2.0.tgz : 404 Not Found
```

**Comment by agardnerIT** [comment]
They have solved the v1.2.0 release issue and your startup script should now "just work". You should be unblocked now.

**Comment by rakshitgondwal** [comment]
@agardnerIT I just tried this out again but I am still facing some errors. 

![Screenshot from 2023-02-13 16-36-34](https://user-images.githubusercontent.com/98955085/218443049-5b2d0760-bd2e-40a2-8a59-3994d64d8c0a.png)
![Screenshot from 2023-02-13 16-41-44](https://user-images.githubusercontent.com/98955085/218443066-8605a615-bfaa-407c-8742-753c6e32b0c4.png)

**Comment by agardnerIT** [comment]
Your errir is indicated in the screenshot - something went wrong with the `ghp_***` token:

1) Did you use your GitHub PAT token not your password?
2) Did your PAT token have full `repo` access?
3) Was the Git repo non-existent before you ran that command?

Keptn v1 requires a brand new and uninitialised repo **every time** you run the command to create a project. Re-using an existing with even a single commit will cause issues.

I cannot recreate this. It works for me:
![image](https://user-images.githubusercontent.com/26523841/218584579-deb06365-2adf-4283-8d42-200d29137ad3.png)

Side note: It looks like there's a new version of `gh` available so perhaps we can also update that during this work?

**Comment by rakshitgondwal** [comment]
Hey @agardnerIT. Yes there was some issue with the github token but it got resolved. I was able to complete the whole tutorial with keptn version `1.2.0` and with gh cli version `2.23.0`. 
I just had one doubt. During the `Attempt to release slow build` in the step 4, the quality checks should fail while running a slow build and the artifact should not be released into production, but while trying out the tutorial, the slow build ran successfully and the artifact was released into production too. 
![Screenshot from 2023-02-14 22-17-53](https://user-images.githubusercontent.com/98955085/218807507-df3f251b-c9d0-4d0f-af2d-c0b24f5d7a23.png)

![Screenshot from 2023-02-14 22-18-19](https://user-images.githubusercontent.com/98955085/218807530-21d79703-bb9f-47c7-ab74-d42af435f993.png)


Also after this step, the application versions should be `v0.1.2` in `qa` and `v0.1.1` in `production`, but I got `v0.1.2` for both.

![Screenshot from 2023-02-14 22-19-14](https://user-images.githubusercontent.com/98955085/218807570-90fd6315-86cc-4e2b-95a7-43d0f8b3a800.png)

**Comment by agardnerIT** [comment]
That is an "issue" with any statically provided threshold. Basically it is because, in your case (and perhaps not in someone else's) YOUR "slow" service was "too quick" - probably because the VM was good.

In other words, your quality gate actually passed - even for a supposedly slow build.

I bet if you look at the response time (sadly you trimmed the table off the screenshot) you'll see it is acceptable to the threshold and therefore Keptn acted correctly.

No bug here.

The "fix" would be to adjust your quality gate threshold(s) in [this file](https://github.com/afzal442/keptn-pg-killerway/blob/master/keptn-end-to-end-delivery/assets/scripts/quality_gated_release/slo.yaml) to bring them back into line.


> TLDR: No bug and you are ready to create the PR and release v1.2.0 🎊
